azure-storage-cpp-lite/src/http/libcurl_http_client.cpp

```diff
@@ -51,9 +55,7 @@ namespace microsoft_azure {
             m_slist = curl_slist_append(m_slist, "Expect:");
             check_code(curl_easy_setopt(m_curl, CURLOPT_HTTPHEADER, m_slist));
 
-            check_code(curl_easy_perform(m_curl));
-
-            return m_code;
+            return curl_easy_perform(m_curl);
```

Comment:
We've had problems in the past where CURL sometimes sets errno when it shouldn't (perhaps even when it returns CURL_OK?  I don't remember for sure.)  That's why check_code resets errno to 0.  I worry if we don't do this here, those bugs might come back

Reply:
but nobody checks errno in this thread as far as I can tell

Reply:
there's no chance that this thread ever gets back to blobfuse?

Reply:
as far as I see, blob_client always uses anync_executor ever for synchronous calls. But if it will make you more confident I will call check_code here as well.

Reply:
Alex we need this merged today :)

Reply:
Also this needs to target dev branch right ?

Reply:
done